authors/apps/comments/views.py

```diff
@@ -0,0 +1,147 @@
+    Handles viewing of comments if not authenticated and\
+    Handles creating comments on an article if authenticated
+    """
+    permission_classes = (IsAuthenticatedOrReadOnly, IsVerifiedUser,)
+    serializer_class = CommentSerializer
+    renderer_classes = (CommentJSONRenderer,)
+
+    def get(self, request, slug):
+        """
+        Handles listing all comments on an article
+
+        :param slug:
+        :return: [comments]
+        """
+        try:
+            article = Articles.objects.get(slug=slug)
+            comments = Comment.objects.all().filter(article_id=article.id)
+            serializer = self.serializer_class(comments, many=True)
+
+            return Response(serializer.data,
+                            status=status.HTTP_200_OK)
+        except Articles.DoesNotExist:
+            return Response({"errors": "Specified Article does not exist."}, status=status.HTTP_404_NOT_FOUND)
+
+    def post(self, request, slug):
+        """
+        Handles creating a new comment on an article
+        :param request:
+        :param slug:
+        :return: comment
+        """
+        try:
+            article = Articles.objects.get(slug=slug)
+            comment = request.data.get('comment', {})
+            serializer = self.serializer_class(data=comment)
+            serializer.is_valid(raise_exception=True)
+            serializer.save(author=request.user, article=article)
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        except Articles.DoesNotExist:
+            return Response({"errors": "Specified Article does not exist."}, status=status.HTTP_404_NOT_FOUND)
+
+
+class RetrieveUpdateDeleteCommentAPIView(APIView):
+    """
+    Handles viewing of a specific comment if not authenticated and\
+    Handles replying to a comment on an article if authenticated
+    Handles Deleting a comment from an article
+    """
+    permission_classes = (IsAuthenticatedOrReadOnly, IsOwnerOrReadOnly, IsVerifiedUser,)
+    serializer_class = CommentSerializer
+    renderer_classes = (CommentJSONRenderer,)
+
+    def get(self, request, *args, **kwargs):
+        """
+        Handles retrieving a specific comment and all their replies
+        :param request:
+        :param args:
+        :param kwargs:
+        :return:
+        """
+        try:
+            article = Articles.objects.get(slug=self.kwargs['slug'])
+        except Articles.DoesNotExist:
+            return Response({"errors": "Specified Article does not exist."}, status=status.HTTP_404_NOT_FOUND)
+        try:
+            comment = Comment.objects.get(pk=self.kwargs['id'])
+            serializer = self.serializer_class(comment)
+            return Response(serializer.data,
+                            status=status.HTTP_200_OK)
+        except Comment.DoesNotExist:
+            return Response({"errors": "Specified Comment does not exist."}, status=status.HTTP_404_NOT_FOUND)
+
+    def post(self, request, *args, **kwargs):
```

Comment:
Can't this already be handled by the `post` method in the class ` RetrieveCommentAPIView` above?

Reply:
The post in:
RetrieveCommentAPIView requires one argument which is the slug to create a new comment.
RetrieveUpdateDeleteCommentAPIView requires two arguments which are the slug and comment id to create a new comment[reply]

Reply:
Each view has it own url that limits what can be sent by the user.